docs/book/src/topics/aso.md

```diff
@@ -79,3 +79,56 @@ More details about how ASO manages CRDs can be found [here](https://azure.github
 
 **Note:** To install the resource for the newly installed CRDs, make sure that the ASO operator has the authentication to install the resources. Refer [authentication in ASO](https://azure.github.io/azure-service-operator/guide/authentication/) for more details.
 An example configuration file and demo for `Azure Cache for Redis` can be found [here](https://github.com/Azure-Samples/azure-service-operator-samples/tree/master/azure-votes-redis).
+
+## Experimental ASO API
+
+CAPZ has begun to experiment with a new flavor of APIs that attempt to address the following limitations of
+the existing CAPZ APIs for advanced use cases:
+
+- A limited set of Azure resource types can be represented.
+- A limited set of Azure resource topologies can be expressed. e.g. Only a single Virtual Network resource can
+  be reconciled for each CAPZ-managed AKS cluster.
+- For each Azure resource type supported by CAPZ, CAPZ generally only uses a single Azure API version to
+  define resources of that type.
+- For each Azure API version known by CAPZ, only a subset of fields defined in that version by the Azure API
+  spec are exposed by the CAPZ API.
+
+This new experimental API defines new AzureASOManagedCluster, AzureASOManagedControlPlane, and
+AzureASOManagedMachinePool resources. An AzureASOManagedCluster might look like this:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureASOManagedCluster
```

Comment:
Are these non-stuttering alternatives better?

- `AzureSOManagedCluster`
- `AzureServiceOperatorManagedCluster`

(I agree that we should stick to having `Azure` as a prefix as a best cloud provider practice here.

Reply:
Of those I think I slightly prefer `AzureServiceOperatorManagedCluster` because `AzureSOManagedCluster` just reads a little less smooth to me. I think I may still like `AzureASOManagedCluster` the best so far because it's a fair bit shorter than `AzureServiceOperatorManagedCluster` but still has a clear tie to ASO. Totally open to changing that later though even if we decide not to now.

Reply:
sgtm